nerdlandbot/commands/space_launches.py

```diff
@@ -111,6 +125,147 @@ def add_embed_field_for_upcominglaunch(self, index, result_json, embed):
         embed.add_field(name = name, value = value, inline = False)
         return
 
+    async def get_percy_data(self):
+        """
+        Returns simplified data for percy pulled from the NASA Mars map API
+        :return: dict containing sol, distance, longitude and latitude (None if failure)
+        """
+        url = PERCY_API_URL
+        full_json = ''
+        try:
+            async with aiohttp.ClientSession() as session:
+                async with session.get(url, headers = {"accept":"application/json"}) as resp:
+                    full_json = await resp.text()
+
+            data = json.loads(full_json)['features'][0]
+
+            percy_data = dict()
+            percy_data['sol'] = data['properties']['sol']
+            percy_data['distance'] = str(data['properties']['dist_km']) + ' km'
+            percy_data['longitude'] = data['geometry']['coordinates'][0]
+            percy_data['latitude'] = data['geometry']['coordinates'][1]
+            return percy_data
+        except Exception as e:
+            error("Failed to load and parse perseverence API: " + 
+                    type(e).__name__ + str(e) +
+                    '--> full_json string head:' + full_json[:100]
+            )
+            return
+    
+    async def get_percy_image(self):
+        """
+        Returns the url to the image of the week
+        :return: string containing the url to image of the week (None if failure)
+        """
+        web_url = PERCY_WEEK_IMG
+        page_html = ''
+        try:
+            async with aiohttp.ClientSession() as session:
+                async with session.get(web_url, headers = {"accept":"text/html"}) as resp:
+                    page_html = await resp.text()
+            
+            soup = BeautifulSoup(page_html, "html.parser")
+            img = soup.find("div", {"class": "main_image"}).find("img")
+            
+            return img['src']
+        except Exception as e:
+            error("Failed to load Perseverance Image of the Week: " + 
+                    type(e).__name__ + str(e) +
+                    '--> page_html string head:' + page_html[:100]
+            )
+            return
+
+
+    async def send_percy_tweet(self, ctx: commands.Context):
+        try:
+            auth = tweepy.AppAuthHandler(self.twitter_creds['key'], self.twitter_creds['secret'])
+            api = tweepy.API(auth)
+            last_status = api.user_timeline(id=PERCY_TWITTER_ID)[0]
```

Comment:
This fetches all tweets and replies, is there a way to limit this to tweets only? I think it would be cleaner to only include tweets and not replies.

Reply:
have limited it to 1